man/gtop.1
.\"
.\" $Id: gtop.1,v 1.1 2003/05/12 11:50:03 vflegel Exp $
.\" Copyright (c) 2003 SIB Swiss Institute of Bioinformatics <pftools@sib.swiss>
.\" Process this file with
.\" groff -man -Tascii <name>
.\" for ascii output or
.\" groff -man -Tps <name>
.\" for postscript output
.\"
.TH GTOP 1 "April 2003" "pftools 2.3" "pftools"
.\" ------------------------------------------------
.\" Name section
.\" ------------------------------------------------
.SH NAME
gtop \- convert a profile in 
.SM GCG
format into
.SM PROSITE
format  
.\" ------------------------------------------------
.\" Synopsis section
.\" ------------------------------------------------
.SH SYNOPSIS
.TP 10
.B gtop
[
.B \-ahls
] [
.B \-E
.I gap_extend
] [
.B \-F
.I rescale
] [
.B \-G
.I gap_open
] [
.B \-O
.I offset
]
.I gcg_profile
|
.B \-
[
.I parameters
]
.\" ------------------------------------------------
.\" Description section
.\" ------------------------------------------------
.SH DESCRIPTION
.B gtop
converts a profile generated by the
.SM GCG
program 
.BR ProfileMake ,
or by a
.SM GCG
compatible profile construction program,
into 
.SM PROSITE
format, and optionally
rescales the profile. 
The result is written to the standard output. 
On input, 
.B gtop
divides all weights of the profile by 100, as
.SM GCG
programs do.
On output, profile scores are multiplied by the rescaling factor
.I F
and rounded to the nearest integer. 
If symmetric gap weighting mode is chosen, the gap opening weights 
are divided 
by 2 and assigned to the 
.IR MI , \ MD , \ IM ,
and
.I DM
transition scores.
If asymmetric gap weighting mode is chosen, the gap opening weights
are left unchanged and assigned to the
.I MI
and
.I MD
transition scores only.
If no argument is specified, the program attempts to read the
.SM GCG
formatted profile from the standard input.
.\" ------------------------------------------------
.\" Options section
.\" ------------------------------------------------
.SH OPTIONS
.\" --- gcg_profile ---
.TP
.I gcg_profile
Input
.SM GCG
profile.
If the filename is replaced by a
.RB ' \- ',
.B gtop
will read the input profile from
.BR stdin .
.\" --- a ---
.TP
.B \-a
Apply asymmetric gap weighting mode (see the
.I description
section).
.\" --- h ---
.TP
.B \-h
Display usage help text.
.\" --- l ---
.TP
.B \-l
Remove output line length limit. Individual lines of the output profile
can exceed a length of 132 characters, removing the need to wrap them over several lines. 
.\" --- s ---
.TP
.B \-s
Apply symmetric gap weighting mode (default) (see the
.I description
section).
.\" --- E ---
.TP
.BI \-E\  gap_extend
Gap extension penalty.
.br
Default: 0.05
.\" --- F ---
.TP
.BI \-F\  rescale
Score rescaling factor.
.br
Default: 100
.\" --- G ---
.TP
.BI \-G\  gap_open
Gap opening penalty.
.br
Default: 4.5
.\" --- O ---
.TP
.BI \-O\  offset
Output score offset. This value is added to all profile match scores
after multiplication by the rescaling factor
.IR F .
.br
Default: 0
.\" ------------------------------------------------
.\" Parameters section
.\" ------------------------------------------------
.SH PARAMETERS
.TP
Note:
for backwards compatibility, release 2.3 of the
.B pftools
package will parse the version 2.2 style parameters, but these are
.I deprecated
and the corresponding option (refer to the
.I options
section) should be used instead.
.TP
E=#
Gap extension penalty.
.br
Use option
.B \-E
instead.
.TP
F=#
Output score rescaling factor.
.br
Use option
.B \-F
instead.
.TP
G=#
Gap opening penalty.
.br
Use option
.B \-G
instead.
.TP
O=#
Output score offset.
.br
Use option
.B \-O
instead.
.\" ------------------------------------------------
.\" Examples section
.\" ------------------------------------------------
.SH EXAMPLES
.TP
(1)
.B gtop
-G 1.0 -E 1.0 -F 50 sh3.gpr | 
.B pfsearch
-far - sh3.seq | 
.B sort
-nr
.IP
First converts the
.SM GCG
profile
.RI ' sh3.gpr '
into
.SM PROSITE
format,
then searches the protein library
.RI ' sh3.seq '
with the reformatted profile 
to generate a sorted list of optimal alignment scores.
The
.BR pfsearch (1)
option
.B \-r
causes alignment scores to be expressed in raw score units. 
.RI ' sh3.seq '
contains 20 SH3 domain-containing protein sequences from
.SM SWISS-PROT
release 32.
.\" ------------------------------------------------
.\" Exit code section
.\" ------------------------------------------------
.SH EXIT CODE
.LP
On successful completion of its task,
.B gtop
will return an exit code of 0. If an error occurs, a diagnostic message will be
output on standard error and the exit code will be different from 0. When conflicting
options where passed to the program but the task could nevertheless be completed, warnings
will be issued on standard error.
.\" ------------------------------------------------
.\" Bugs section
.\" ------------------------------------------------
.SH BUGS
.B gtop
defines 
.I SCORE=0; N_SCORE=0;
for the mandatory cut-off level zero. These parameters should be changed 
to more reasonable values using a text editor, prior to profile search
operations.
.\" ------------------------------------------------
.\" See also section
.\" ------------------------------------------------
.SH "SEE ALSO"
.BR pfsearch (1),
.BR pfscan (1),
.BR pfmake (1),
.BR sort (1)
.\" ------------------------------------------------
.\" Author section
.\" ------------------------------------------------
.SH AUTHOR
The
.B pftools
package was developed by Philipp Bucher.
.br
Any comments or suggestions should be addressed to <pftools@sib.swiss>.


src/Fortran/gtop.f
*       Program gtop 
*----------------------------------------------------------------------*     
* $Id: gtop.f,v 2.9 2003/11/28 11:53:33 vflegel Exp $
*----------------------------------------------------------------------*     
*       Function: Reformats profiles: in-fmt=GRIBSKOV / out-fmt=PROSITE    
*       Author:   Philipp Bucher
*       Contact:  pftools@sib.swiss
*       Version:  File under developpment for release 2.3
*----------------------------------------------------------------------*     
*
* DATA
*----------------------------------------------------------------------*     

* array dimensions and I/O units
      
      Include          'ardim.f' 

      Parameter        (NOUT=   6)    
      Parameter        (NGPR=  11)    

* profile and sequence fields

      Character*512     FGPR

      Include          'psdat.f'
      Include          'gsdat.f'
      Include          'djdat.f'
      Include          'nodat.f'
      Include          'codat.f'
      Include          'pfdat.f'
      Include          'dfdat.f'
      Include          'sterr.f'

* command line options and parameters

      Logical           LSYM
      Logical           LLLT
      Logical           LRNM

* initialization of controlled vocabularies

      Include          'cvini.f' 

*----------------------------------------------------------------------*     
* INPUT SECTION 
*----------------------------------------------------------------------*     

      IRC=0
      FGPR='-'
      BLOG=0.0
      LRNM=.FALSE.

* read command line

      RL=NLOW
      Call Repar
     *   (FGPR,LSYM,LLLT,RG,RE,RF,RO,IRC)
      If(IRC.NE.0) then
         Write(NERR,'(/,
     *      ''gtop 2.3 revision 5.d'',//
     *      ''Usage: gtop [ -aslhGEFO ] gcg-profile | - '' 
     *      ''[ parameters ]'',/
     *      )')
         Write(NERR,'(
     *      ''   options:'',/,
     *      ''    -a: apply asymmetric gap weighting mode.'',/
     *      ''    -s: apply symmetric gap weighting mode (default).'',/
     *      ''    -l: do not impose limit on line length.'',/
     *      ''    -h: print usage help text.'',/
     *      ''    -G<value>:'',/
     *      ''        gap opening penalty (default: 4.5).'',/
     *      ''    -E<value>:'',/
     *      ''        gap extension penalty (default: 0.05).'',/
     *      ''    -F<value>:'',/
     *      ''        rescaling factor (default: 100).'',/
     *      ''    -O<value>:'',/
     *      ''        output score offset (default: 0). '',/
     *      ''        Added to match scores after multiplication by '',
     *      ''the rescaling factor F.'',/
     *      )')
         Write(NERR,'(
     *      ''   valid (but deprecated) parameters are:'',//,
     *      ''    [G=gap-weigth]              use option -G instead'',/,
     *      ''    [E=gap-extension-weight]    use option -E instead'',/,
     *      ''    [F=output-score-multiplier] use option -F instead'',/
     *      ''    [O=output-score-offset]     use option -O instead'',/
     *      )')
         Call Exit(IRC)
      End if

* read profile

      Call REGPR
     *   (NGPR,FGPR,
     *   RG,RE,RF,RO,LSYM,
     *   CPID,CPAC,CPDE,NABC,CABC,LPRF,LPCI,
     *   CDIS,JDIP,MDIS,NDIP,
     *   CNOR,JNOP,JNOR,MNOR,NNOR,NNPR,CNTX,RNOP,
     *   JCUT,MCLE,CCUT,ICUT,JCNM,RCUT,MCUT, 
     *   IDMP,CHIP,IIPP,CHMP,IMPP,
     *   CHID,IIPD,CHMD,IMPD,
     *   IRC)

      If(IRC.NE.0) go to 100

      CPDT=' '
      LHDR=0
      LFTR=0

      Call WRPRF
     *   (NOUT,LLLT,LRNM,
     *   CPID,CPAC,CPDT,CPDE,LHDR,CHDR,LFTR,CFTR,NABC,CABC,LPRF,LPCI,
     *   CDIS,JDIP,MDIS,NDIP,
     *   CNOR,JNOP,JNOR,MNOR,NNOR,NNPR,CNTX,RNOP,
     *   JCUT,MCLE,CCUT,ICUT,JCNM,RCUT,MCUT, 
     *   IDMP,CHIP,IIPP,CHMP,IMPP,
     *   BLOG,FABC,P0,
     *   CHID,IIPD,CHMD,IMPD,
     *   IRC)

 100  Call Exit(IRC)
      End
*----------------------------------------------------------------------*     
      Subroutine Repar         
     *   (FGPR,LSYM,LLLT,RG,RE,RF,RO,IRC)

      Character*512     CPAR
      Character*(*)     FGPR 
      
      Logical           LSYM
      Logical           LLLT
      
      LSYM=.TRUE.
      LLLT=.TRUE.
      RG=4.5
      RE=0.05
      RF=100
      RO=0
      
      FGPR=' '

      N1=Iargc()
      
      K1=0
      I2=1
      Do  50 I1=1,N1
         Call GetArg(I2,CPAR)
         If     (CPAR(1:1).EQ.'-'
     *      .AND.CPAR(2:2).NE.' '.AND.K1.EQ.0) then
            If(Index(CPAR,'h').NE.0)go to 900
            If(Index(CPAR,'a').NE.0) LSYM=.FALSE.
            If(Index(CPAR,'l').NE.0) LLLT=.FALSE.
            If(Index(CPAR,'G').NE.0) then
               If(CPAR(3:3).NE.' ') then
                  Read(CPAR(3:),*,Err=900) RG
               Else
                  I2=I2+1
                  Call GetArg(I2,CPAR)
                  Read(CPAR,*,Err=900) RG
               End if
            End if
            If(Index(CPAR,'E').NE.0) then
               If(CPAR(3:3).NE.' ') then
                  Read(CPAR(3:),*,Err=900) RE
               Else
                  I2=I2+1
                  Call GetArg(I2,CPAR)
                  Read(CPAR,*,Err=900) RE
               End if
            End if
            If(Index(CPAR,'F').NE.0) then
               If(CPAR(3:3).NE.' ') then
                  Read(CPAR(3:),*,Err=900) RF
               Else
                  I2=I2+1
                  Call GetArg(I2,CPAR)
                  Read(CPAR,*,Err=900) RF
               End if
            End if
            If(Index(CPAR,'O').NE.0) then
               If(CPAR(3:3).NE.' ') then
                  Read(CPAR(3:),*,Err=900) RO
               Else
                  I2=I2+1
                  Call GetArg(I2,CPAR)
                  Read(CPAR,*,Err=900) RO
               End if
            End if
         Else if(K1.EQ.0) then
            K1=K1+1
            FGPR=CPAR
         Else
            If     (CPAR(1:2).EQ.'G=') then
               Read(CPAR(3:),*,Err=900) RG
            Else if(CPAR(1:2).EQ.'E=') then
               Read(CPAR(3:),*,Err=900) RE
            Else if(CPAR(1:2).EQ.'F=') then
               Read(CPAR(3:),*,Err=900) RF
            Else if(CPAR(1:2).EQ.'O=') then
               Read(CPAR(3:),*,Err=900) RO
            End if 
         End if
         I2=I2+1
         If(I2.GT.N1) Go to 60
 50   Continue

 60   If(K1.NE.1) Go to 900 
 100  Return
 900  IRC=1
      Go to 100
      End 
*----------------------------------------------------------------------*     
      Include          'lblnk.f'
      Include          'regpr.f'
      Include          'wrprf.f'
